packages/loopring_v3/contracts/core/iface/ExchangeData.sol

```diff
@@ -28,6 +28,12 @@ library ExchangeData
         AMM_UPDATE
     }
 
+    enum ApprovalType
+    {
+     USER,
+     AMM_POOL
```

Comment:
Exchange should not know anything about AMM, what if we have another application, shall we modify the exchange contract with a new ApprovalType? No!